win32ss/user/ntuser/defwnd.c

```diff
@@ -792,32 +789,27 @@ IntDefWindowProc(
                co_IntSendMessage(UserHMGetHandle(Wnd), WM_CONTEXTMENU, (WPARAM)UserHMGetHandle(Wnd), MAKELPARAM(-1, -1));
             }
          }
-         if (IS_KEY_DOWN(gafAsyncKeyState, VK_LWIN) || IS_KEY_DOWN(gafAsyncKeyState, VK_RWIN))
+         if (g_bWindowSnapEnabled && (IS_KEY_DOWN(gafAsyncKeyState, VK_LWIN) || IS_KEY_DOWN(gafAsyncKeyState, VK_RWIN)))
          {
+            BOOL IsTaskBar;
+            DWORD StyleTB;
+            DWORD ExStyleTB;
             HWND hwndTop = UserGetForegroundWindow();
             PWND topWnd = UserGetWindowObject(hwndTop);
 
-            /* Test for typical TaskBar ExStyle Values */
-            ExStyle = (topWnd->ExStyle & WS_EX_TOOLWINDOW);
-            TRACE("ExStyle is '%x'.\n", ExStyle);
-
-            /* Test for typical TaskBar Style Values */
-            Style = (topWnd->style & (WS_POPUP | WS_VISIBLE |
-                        WS_CLIPSIBLINGS | WS_CLIPCHILDREN));
-            TRACE("Style is '%x'.\n", Style);
-
-            /* Test for masked typical TaskBar Style and ExStyles to detect TaskBar */
-            IsTaskBar = (Style == (WS_POPUP | WS_VISIBLE | WS_CLIPSIBLINGS | WS_CLIPCHILDREN))
-                        && (ExStyle == WS_EX_TOOLWINDOW);
-            TRACE("This %s the TaskBar.\n", IsTaskBar ? "is" : "is not");
+            // We want to forbid snapping operations on the TaskBar
```

Comment:
Question, for here and the other file:
Can you explain what this "forbidding snapping operations on the taskbar" is all about?
When you test on Windows, by moving the taskbar to e.g. the top, the left or the right of the screen (instead of the standard bottom position), you will notice that the snapping action is forbidden only at the bottom, independently of where the taskbar is.

![image](https://user-images.githubusercontent.com/1969829/218276766-7e46ed62-2c81-47c0-b985-c9a2db31f6d2.png)

Dragging any window to the bottom (of the "workarea"/screen) won't make the snapping work.

Reply:
Although your question is totally out of the scope of this PR, I still want to answer your question: see the resolved https://jira.reactos.org/browse/CORE-16289 to understand what this code/fix of Doug was about.